tests/src/main/java/com/hazelcast/stabilizer/tests/map/MapHeapHogWarmupTest.java

```diff
@@ -0,0 +1,136 @@
+package com.hazelcast.stabilizer.tests.map;
+
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.core.IMap;
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.logging.Logger;
+import com.hazelcast.stabilizer.test.TestContext;
+import com.hazelcast.stabilizer.test.annotations.Run;
+import com.hazelcast.stabilizer.test.annotations.Setup;
+import com.hazelcast.stabilizer.test.annotations.Verify;
+import com.hazelcast.stabilizer.test.annotations.Warmup;
+import com.hazelcast.stabilizer.test.utils.ThreadSpawner;
+
+import java.util.concurrent.TimeUnit;
+
+import static com.hazelcast.stabilizer.test.utils.TestUtils.humanReadableByteCount;
+import static com.hazelcast.stabilizer.tests.helpers.HazelcastTestUtils.isClient;
+import static com.hazelcast.stabilizer.tests.helpers.HazelcastTestUtils.nextKeyOwnedBy;
+import static com.hazelcast.stabilizer.utils.CommonUtils.sleepSeconds;
+import static java.lang.String.format;
+
+/**
+ * Fills up the cluster to a defined heap usage factor during warmup phase.
+ * <p/>
+ * This tests intentionally has an empty run phase.
+ */
+public class MapHeapHogWarmupTest {
+
+    private static final int MIN_CLUSTER_SIZE = 3;
+    private static final long APPROX_ENTRY_BYTES_SIZE = 238;
+    private static final ILogger LOGGER = Logger.getLogger(MapHeapHogWarmupTest.class);
+
+    // properties
+    public String basename = this.getClass().getSimpleName();
+    public int threadCount = 10;
+    public int ttlHours = 24;
+    public double approxHeapUsageFactor = 0.9;
+
+    private HazelcastInstance targetInstance;
+    private ThreadSpawner threadSpawner;
+    private IMap<Long, Long> map;
+
+    private long maxEntriesPerThread;
+
+    @Setup
+    public void setup(TestContext testContext) {
+        targetInstance = testContext.getTargetInstance();
+        threadSpawner = new ThreadSpawner(testContext.getTestId());
+        map = targetInstance.getMap(basename);
+
+        // calculate how many entries we have to insert per member and thread
+        long free = Runtime.getRuntime().freeMemory();
+        long total = Runtime.getRuntime().totalMemory();
+        long max = Runtime.getRuntime().maxMemory();
+
+        long used = total - free;
+        long totalFree = max - used;
+
+        long maxLocalEntries = (long) ((totalFree / APPROX_ENTRY_BYTES_SIZE) * approxHeapUsageFactor);
+        maxEntriesPerThread = (long) ((double) maxLocalEntries / (double) threadCount);
+    }
+
+    @Warmup()
+    public void warmup() {
+        if (isClient(targetInstance)) {
+            return;
+        }
+
+        // wait for the cluster to build up
+        while (targetInstance.getCluster().getMembers().size() != MIN_CLUSTER_SIZE) {
```

Comment:
I think we should have some kind of timeout. You don't want to wait till the end of time.